client/app/vps/vps-header.controller.js

```diff
@@ -0,0 +1,88 @@
+class VpsHeaderCtrl {
+    constructor ($rootScope, $stateParams, $translate, CloudMessage, Notification, STOP_NOTIFICATION_USER_PREF, VpsService) {
+        this.$rootScope = $rootScope;
+        this.$stateParams = $stateParams;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+        this.Notification = Notification;
+        this.STOP_NOTIFICATION_USER_PREF = STOP_NOTIFICATION_USER_PREF;
+        this.serviceName = $stateParams.serviceName;
+        this.description = $stateParams.displayName || $stateParams.serviceName;
+        this.VpsService = VpsService;
+        this.loaders = {
+            init: false
+        };
+        this.stopNotification = {
+            autoRenew: true,
+            ipV6: true
+        };
+    }
+
+    $onInit () {
+        this.loaders.init = true;
+        this.$rootScope.$on("changeDescription", (event, data) => {
+            this.description = data;
+        });
+        this.VpsService.getSelectedVps(this.serviceName)
+            .then(vps => {
+                this.description = vps.displayName;
+                this.checkMessages(vps);
+            })
+            .catch(() => this.CloudMessage.error(this.$translate.instant("vps_dashboard_loading_error")))
+            .finally(() => { this.loaders.init = false });
+    }
+
+    checkMessages (vps) {
+        this.isExpired(vps);
+        this.isInRescueMode(vps.netbootMode);
+        this.checkIfStopNotification("ipV6", true, vps);
```

Comment:
It is sad that all of this is not in a service but I understand you have no time for that

Reply:
Some of function there would be deleted there ! So that's why I have just imported some of then.